iothub/service/src/HttpTransportSettings.cs

```diff
@@ -19,6 +19,10 @@ public HttpTransportSettings()
             Proxy = DefaultWebProxySettings.Instance;
         }
 
+        // -1 is the default value that this SDK had inadvertantly set by not configuring this value in the past. Changing this default may be a
+        // breaking change considering it impacts the performance of the SDK
+        internal const int DefaultConnectionLeaseTimeoutMilliseconds = -1;
```

Comment:
This feels worth discussing to me. By defaulting to -1, we avoid any possible breaking change, because this is the default value we already had by not setting this ServicePoint property. However, it also seems reasonable to me to make this value 1 minute or something just so that users don't run into this issue by default when they failover their IoT Hub. The "breaking change" would only be a small performance hit as the http client's check for DNS changes every minute rather than never checking.